cocotb/outcomes.py

```diff
@@ -44,14 +46,14 @@ def __repr__(self):
 
 
 class Error(Outcome):
-    def __init__(self, error):
-        self.error = error
+    def __init__(self, *exc_info):
+        self.exc_info = exc_info
```

Comment:
Would it make more sense to have different versions of `Error` for python 2 and 3? AFAICT, these changes have no effect on 3, and just make things harder to read.